packages/plugin-ext/src/main/browser/webview/webview.ts

```diff
@@ -13,316 +13,571 @@
+                theme => `background-image: url(${theme.id === BuiltinThemeProvider.lightTheme.id ? lightIconUrl : darkIconUrl});`
+            ));
+            this.title.iconClass = `theia-webview-icon ${iconClass}`;
+        } else {
+            this.title.iconClass = '';
+        }
+    }
+
+    setHTML(value: string): void {
+        this.html = this.preprocessHtml(value);
+        this.doUpdateContent();
+    }
 
-        this.updateSandboxAttribute(newFrame);
+    protected preprocessHtml(value: string): string {
+        return value
+            .replace(/(["'])theia-resource:(\/\/([^\s\/'"]+?)(?=\/))?([^\s'"]+?)(["'])/gi, (_, startQuote, _1, scheme, path, endQuote) => {
+                if (scheme) {
+                    return `${startQuote}${this.externalEndpoint}/theia-resource/${scheme}${path}${endQuote}`;
+                }
+                return `${startQuote}${this.externalEndpoint}/theia-resource/file${path}${endQuote}`;
+            });
     }
 
     protected onActivateRequest(msg: Message): void {
         super.onActivateRequest(msg);
-        // restore scrolling if there was one
-        if (this.scrollY > 0) {
-            this.iframe.contentWindow!.scrollTo({ top: this.scrollY });
-        }
+        this.focus();
+    }
+
+    focus(): void {
         this.node.focus();
-        // unblock messages
-        this.readyToReceiveMessage = true;
+        if (this.element) {
+            this.doSend('focus');
+        }
     }
 
-    // block messages
-    protected onBeforeShow(msg: Message): void {
-        this.readyToReceiveMessage = false;
+    reload(): void {
+        this.doUpdateContent();
     }
 
-    protected onBeforeHide(msg: Message): void {
-        // persist scrolling
-        if (this.iframe.contentWindow) {
-            this.scrollY = this.iframe.contentWindow.scrollY;
-        }
-        super.onBeforeHide(msg);
+    protected style(): void {
+        const { styles, activeTheme } = this.themeDataProvider.getThemeData();
+        this.doSend('styles', { styles, activeTheme });
     }
 
-    public reloadFrame(): void {
-        if (!this.iframe || !this.iframe.contentDocument || !this.iframe.contentDocument.documentElement) {
-            return;
-        }
-        this.setHTML(this.iframe.contentDocument.documentElement.innerHTML);
+    protected dispatchKeyDown(event: KeyboardEventInit): void {
+        // Create a fake KeyboardEvent from the data provided
+        const emulatedKeyboardEvent = new KeyboardEvent('keydown', event);
+        // Force override the target
+        Object.defineProperty(emulatedKeyboardEvent, 'target', {
+            get: () => this.element,
+        });
+        // And re-dispatch
+        this.keybindings.run(emulatedKeyboardEvent);
     }
 
-    private updateSandboxAttribute(element: HTMLElement, isAllowScript?: boolean): void {
-        if (!element) {
-            return;
+    protected openLink(link: URI): void {
+        if (this.isSupportedLink(link)) {
```

Comment:
We need to detect if the link is pointing to a resource in the workspace.
I.e. getting a link like `https://<web-view-origin-host>/webview/theia-resource/file/workspace/theia/doc/Developing.md#quick-start` should be translated to open
`file://workspace/theia/doc/Developing.md#quick-start`.

Reply:
It does not work because of https://github.com/eclipse-theia/theia/pull/6465#issuecomment-554989037

Reply:
We also should move to latest markdown built-in extension. Previous version expect `file://` and `vscode-resource://` href, but it is not possible since they are get rewritten to origin based href. It is the same in VS Code.

Reply:
I've managed to publish the latest markdown built-in extension from November:
```
wget https://registry.npmjs.org/@theia/vscode-builtin-markdown-language-feature/-/vscode-builtin-markdown-language-feature-0.3.0-next.c9a2de2d9c.tgz
wget https://registry.npmjs.org/@theia/vscode-builtin-markdown-language-features/-/vscode-builtin-markdown-language-features-0.3.0-next.c9a2de2d9c.tgz
```

It still cannot handle relative links properly. I've checked it is the same in VS Code online. I will do your suggestion for now.

Reply:
Pushed to resolve such links to file links: https://github.com/eclipse-theia/theia/pull/6465/commits/453e1e93e79fcb3e6d2fc3a53e2edd9273c11e01

Ideally it should be fixed in the markdown extension though.